ops/src/purge/purger.rs

```diff
@@ -433,3 +439,374 @@ pub async fn start_mint_purge<P: DatabasePool>(args: Args, db: P, rpc: Rpc) -> R
+            &mpl_bubblegum::ID,
+        );
+        let asset_bytes = asset.to_bytes().to_vec();
+
+        let node_path = calculate_node_path(leaf_idx, tree.clone(), &conn).await?;
+
+        let multi_txn = conn.begin().await?;
+
+        let asset_data = asset_data::Entity::delete_by_id(asset_bytes.clone())
+            .exec(&multi_txn)
+            .await?;
+
+        debug!("asset_data DeleteResult: {:?}", asset_data);
+
+        let asset = asset::Entity::delete_by_id(asset_bytes.clone())
+            .exec(&multi_txn)
+            .await?;
+
+        debug!("asset DeleteResult: {:?}", asset);
+
+        let asset_creators = asset_creators::Entity::delete_many()
+            .filter(asset_creators::Column::AssetId.eq(asset_bytes.clone()))
+            .exec(&multi_txn)
+            .await?;
+
+        debug!("asset_creators DeleteResult: {:?}", asset_creators);
+
+        let asset_authority = asset_authority::Entity::delete_many()
+            .filter(asset_authority::Column::AssetId.eq(asset_bytes.clone()))
+            .exec(&multi_txn)
+            .await?;
+
+        debug!("asset_authority DeleteResult: {:?}", asset_authority);
+
+        let asset_grouping = asset_grouping::Entity::delete_many()
+            .filter(asset_grouping::Column::AssetId.eq(asset_bytes))
+            .exec(&multi_txn)
+            .await?;
+
+        debug!("asset_grouping DeleteResult: {:?}", asset_grouping);
+
+        let cl_items = cl_items::Entity::delete_many()
+            .filter(cl_items::Column::Tree.eq(tree.clone()))
+            .filter(cl_items::Column::NodeIdx.is_in(node_path))
+            .filter(cl_items::Column::Seq.lte(tx_seq))
+            .exec(&multi_txn)
+            .await?;
+
+        debug!("cl_items DeleteResult: {:?}", cl_items);
+
+        // Remove all txs for the asset from cl_audits_v2 so that the asset can be re-indexed by the backfiller
+        let cl_audits_v2 = cl_audits_v2::Entity::delete_many()
+            .filter(cl_audits_v2::Column::LeafIdx.eq(leaf_idx))
+            .filter(cl_audits_v2::Column::Tree.eq(tree.clone()))
+            .exec(&multi_txn)
+            .await?;
+        debug!("cl_audits_v2 DeleteResult: {:?}", cl_audits_v2);
+
+        multi_txn.commit().await?;
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Parser, Clone)]
+pub struct CnftArgs {
+    /// The list of trees to verify. If not specified, all trees will be crawled.
+    #[arg(long, env, use_value_delimiter = true)]
+    pub only_trees: Option<Vec<String>>,
+    #[clap(flatten)]
+    pub purge_args: Args,
+}
+
+pub async fn start_cnft_purge<P: DatabasePool>(args: CnftArgs, db: P, rpc: Rpc) -> Result<()> {
+    let start = tokio::time::Instant::now();
+
+    let purge_worker_count = args.purge_args.purge_worker_count as usize;
+    let only_trees = args.only_trees.map(|trees| {
+        trees
+            .into_iter()
+            .map(|tree| bs58::decode(tree).into_vec().unwrap())
+            .collect()
```

Comment:
```suggestion
    let only_trees = args.only_trees.map(|trees| {
        trees
            .into_iter()
            .map(|tree| bs58::decode(tree).into_vec())
            .collect::<Result<Vec<u8>>>()
```

You can collect the results instead of unwrap in the map.